storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -400,14 +413,9 @@ func (s *fileSystemSource) Read(
 		xlog.NewField("concurrency", s.opts.NumProcessors()),
 		xlog.NewField("metadataOnly", blockRetriever != nil),
 	).Infof("filesystem bootstrapper bootstrapping shards for ranges")
-	readerPool := newReaderPool(func() fs.FileSetReader {
-		return s.newReaderFn(
-			s.filePathPrefix,
-			s.dataReaderBufferSize,
-			s.infoReaderBufferSize,
-			s.opts.ResultOptions().DatabaseBlockOptions().BytesPool(),
-			s.decodingOpts,
-		)
+	bytesPool := s.opts.ResultOptions().DatabaseBlockOptions().BytesPool()
+	readerPool := newReaderPool(func() (fs.FileSetReader, error) {
+		return s.newReaderFn(bytesPool, s.fsopts)
```

Comment:
Any reason to have this here instead of `Options`?

Reply:
Yeah you don't really want to allocate and keep around readers outside of the bootstrapping process, hence why its created on demand each time.